core/src/saros/activities/TextSelectionActivity.java

```diff
@@ -46,15 +46,29 @@
   @XStreamAsAttribute
   private final int endInLineOffset;
 
+  @XStreamAlias("bs")
+  private final boolean isBackwardsSelection;
+
+  /**
+   * Calls {@link #TextSelectionActivity(User, TextSelection, boolean, IFile)} with <code>
+   * isBackwardsSelection=false</code>.
+   */
+  public TextSelectionActivity(User source, TextSelection selection, IFile file) {
```

Comment:
I am missing some class description here. You could also implement backwards selection by allowing a reverse order of the offsets which would actually reflects a backward selection. But I guess you have not done this to avoid changes in the Eclipse implementation.

It still would be nice to describe in the class summary what this activity contains, e.g Eclipse has no usage for the isBackwardSelection flag and other IDE could also not support this logic.

Reply:
Ok I checked the other patch set and it seems I have a different understanding of what a backward selection is all about. Can you please move the definition (can be found in the IntelliJ part) into this class ?

Reply:
> Can you please move the definition (can be found in the IntelliJ part) into this class ?

Isn't this already defined in the javadoc for [`isBackwardsSelection`](https://github.com/saros-project/saros/pull/966/files#diff-a09aa94b56b0d656967b9d2de45a8a75R112).

> I am missing some class description here. You could also implement backwards selection by allowing a reverse order of the offsets which would actually reflects a backward selection. But I guess you have not done this to avoid changes in the Eclipse implementation.

Yes, I could have also implemented backwards selections by allowing the selection start to be after the selection end. But, as you already pointed out, maybe not all IDEs (or all Saros implementations) support backwards selections. With the current implementation, if the local IDE and Saros implementation support backwards selections, they will be displayed as such. If not, they will be displayed as normal (forwards) selections.

I did not adjust the Eclipse implementation as I am not familiar with the API for requesting caret positions and did not want to invest time into it (as it is very much a "nice to have"/optional feature). But I will open a feature request to also implement it for Saros/E so that we don't forget about it.